certbot-apache/certbot_apache/constants.py

```diff
@@ -62,9 +62,9 @@
     vhost_files="*.conf",
     logs_root="/var/log/apache2",
     version_cmd=['/usr/sbin/apache2', '-v'],
-    define_cmd=['apache2ctl', 'virtualhosts'],
-    restart_cmd=['apache2ctl', 'graceful'],
-    conftest_cmd=['apache2ctl', 'configtest'],
+    define_cmd=['/usr/sbin/apache2', '-S'],
```

Comment:
A possibly working but extremely dirty hack may be using:
```
    version_cmd=['/bin/sh', '-c', '. /etc/conf.d/apache2; apache2 ${APACHE2_OPTS} -v'],
    define_cmd=['/bin/sh', '-c', '. /etc/conf.d/apache2; apache2 ${APACHE2_OPTS} -S'],
    restart_cmd=['/bin/sh', '-c', '. /etc/conf.d/apache2; apache2 ${APACHE2_OPTS} -k graceful'],
    conftest_cmd=['/bin/sh', '-c', '. /etc/conf.d/apache2; apache2 ${APACHE2_OPTS} -t'],
```
In short, this "config" plays inception with `subprocess.Popen` (which, in the end, executes what is defined in `CLI_DEFAULTS_GENTOO`) by making it start a shell that itself starts a shell that sources `/etc/conf.d/apache2` and then executes the apache2 command with the sourced `APACHE2_OPTS` (just like `/etc/init.d/apache2` and `/usr/lib/systemd/system/apache2.service` do).

It passed my tests in a VM but since I have no complete test setup for this project running I cannot confirm that it will work in every situation so if you (@osirisinferi) can take over the formal testing, then we may have a solution that will work for both of us AND enables everyone to use the automatic installation.

But still, it would be a dirty hack that should be removed as soon as certbot supports `CLI_DEFAULTS` depending on the used init system.